src/index.ts

```diff
@@ -53,27 +53,29 @@ window.addEventListener('load', () => {
   });
 });
 
-function onDBOpenSuccess(e, imgElement, shareBtn) {
+function onDBOpenSuccess(e: Event, imgElement: HTMLImageElement, shareBtn: HTMLButtonElement) {
   console.log('Database initialized');
 
-  const db = e.target.result;
+  const db = (e.target as IDBRequest).result;
   const t = db.transaction(objStoreName, 'readonly').objectStore(objStoreName).get(mainImageName);
-  t.addEventListener('success', (e) => {
+  t.addEventListener('success', (e: Event) => {
     console.log('get success:', e)
-    const b = e.target.result;
+    const b = (e.target as IDBRequest).result;
     if (!b) return;
     imgElement.src = window.URL.createObjectURL(b);
     displayIfShareEnabled(shareBtn, b);
   });
-  t.addEventListener('error', (e) => {
+  t.addEventListener('error', (e: Event) => {
     console.log('get error:', e)
   });
 }
 
-async function onFileInputChange(e, dbOpenRequest, imgElement, shareBtn) {
-  console.log(e.target.value);
-  if (e.target.files.length === 0) return;
-  const f = e.target.files[0]; // TODO: null-check
+async function onFileInputChange(e: Event, dbOpenRequest: IDBOpenDBRequest, imgElement: HTMLImageElement, shareBtn: HTMLButtonElement) {
+  const inputElement = e.target as HTMLInputElement;
+
+  console.log(inputElement.value);
+  if (!inputElement.files || !inputElement.files.length) return;
+  const f = inputElement.files[0]; // TODO: null-check
```

Comment:
is it not possible for `f` to be null here? I would've expected typescript to complain

Reply:
The TypeScript definition doesn't allow null to be returned when accessing files by index: https://github.com/microsoft/TypeScript/blob/main/lib/lib.dom.d.ts#L5224-L5228. Seems to make sense from the spec? https://w3c.github.io/FileAPI/#filelist-section

Reply:
I've avoided making changes in this PR unrelated to enabling TypeScript - thus the length check. I'm not seeing where Blink would return a NULL entry in the array.